File: ext/owl/http/responseinterface.zep.c

#ifdef HAVE_CONFIG_H
#include "../../ext_config.h"
#endif

#include <php.h>
#include "../../php_ext.h"
#include "../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Owl_Http_ResponseInterface) {

	ZEPHIR_REGISTER_INTERFACE(Owl\\Http, ResponseInterface, owl, http_responseinterface, owl_http_responseinterface_method_entry);

	return SUCCESS;

}

ZEPHIR_DOC_METHOD(Owl_Http_ResponseInterface, send);

ZEPHIR_DOC_METHOD(Owl_Http_ResponseInterface, getContent);

ZEPHIR_DOC_METHOD(Owl_Http_ResponseInterface, setCode);

ZEPHIR_DOC_METHOD(Owl_Http_ResponseInterface, getCode);

File: ide/Owl/http/ResponseInterface.zep.php
<?php

namespace Owl\Http;


interface ResponseInterface
{

    /**
     * @return bool 
     */
    public function send();

    /**
     * @return string 
     */
    public function getContent();

    /**
     * @param int $code 
     */
    public function setStatusCode($code);

    /**
     * @return int 
     */
    public function getStatusCode();

}
